lib/ansible/module_utils/saslprep.py

```diff
@@ -1,16 +1,5 @@
-# Copyright 2016-present MongoDB, Inc.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-# http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
+# Copyright (c) 2019 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
```

Comment:
This code is copyright to MongoDB, Inc.  I don't think you can simply switch out copyright and license like this.

My recommendation on IRC was that `pymongo` should just be made a soft dependency of the module, and only required as needed.

Reply:
@sivel Yes, I should have kept the copyright.

I don't want to add `pymongo` as a dependency this makes no sense in this case

1. I can keep the copyright and use dual license in this case (it is permitted by Apache2 / GPL3 AFAIK) 
2. I can use another implementation licensed under BSDm keeping the copyright and replacing the license (it is permitted by BSD / GPL3 AFAIK)
3. I can rewrite this one from scratch...

My personal preference is (2) if (1) is not an option. Are you ok with (1) or (2)?

Reply:
@Andersson007

Reply:
hey:) I saw and tried to ask @sivel in irc for looking at this yesterday. Probably we should try to do it again today later.
(In my opinion, if I could I tried to rewrite this actually from scratch with a suitable license)

Reply:
Use of any license other than BSD for `module_utils` is going to require at minimum core team agreement.  As far as dual licensing is concerned, we do not accept dual licenses.  You would need to convince the original authors to accept a new license.  Any work related to this is likely to require discussions with the Red Hat legal team as well.

In short, our recommendation would be to find another bit of code that is BSD licensed, or to not bundle this code and require a new external dependency for modules needing this functionality.

I think your preference for the 2nd option would be preferable. Finding another implementation already licensed under BSD.

Reply:
We could look at using [Passlib](https://passlib.readthedocs.io/en/stable/) it is BSD licensed.

Reply:
> We could look at using Passlib it is BSD licensed.
@andytom @sivel
Yep, I was going to take passlib implementation, it needs some work, but it's fine, thanks!

Reply:
@sivel 
I've updated `saslprep.py` module sources (now under BSD license).